library/x509_crt.c

```diff
@@ -2524,6 +2529,142 @@ static int x509_crt_verify_chain(
     }
 }
 
+#ifdef _WIN32
+#ifdef _MSC_VER
+#pragma comment(lib, "ws2_32.lib")
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#elif (defined(__MINGW32__) || defined(__MINGW64__)) && _WIN32_WINNT >= 0x0600
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#endif
+#elif defined(__sun)
+/* Solaris requires -lsocket -lnsl for inet_pton() */
+#elif defined(__has_include)
+#if __has_include(<sys/socket.h>)
+#include <sys/socket.h>
+#endif
+#if __has_include(<arpa/inet.h>)
+#include <arpa/inet.h>
+#endif
+#endif
+
+/* Use whether or not AF_INET6 is defined to indicate whether or not to use
+ * the platform inet_pton() or a local implementation (below).  The local
+ * implementation may be used even in cases where the platform provides
+ * inet_pton(), e.g. when there are different includes required and/or the
+ * platform implementation requires dependencies on additional libraries.
+ * Specifically, Windows requires custom includes and additional link
+ * dependencies, and Solaris requires additional link dependencies.
+ * Also, as a coarse heuristic, use the local implementation if the compiler
+ * does not support __has_include(), or if the definition of AF_INET6 is not
+ * provided by headers included (or not) via __has_include() above. */
+#ifndef AF_INET6
```

Comment:
On our CI, we only run tests on Linux, FreeBSD and Windows (Visual Studio 2013+ and MinGW, I don't know offhand what MinGW version(s)). I think all of them have IPv6 support. So are we ever testing the implementation of `inet_pton` here? If not we should arrange some build where those functions are tested.

Reply:
@AndrzejKurek wrote:
> And lastly - if you feel overwhelmed / you feel like you won't have the time to commit to this PR, let us know - we'll try to take it over. Thanks again!

I would appreciate if you would handle all additional increases in scope or testing.  Thank you.